File: OneQuestionPerDay/387/first_unique_character_in_a_string.go
package main

func firstUniqChar(s string) int {
	arr := [26]int{}

	for i := 0;i < len(s);i++ {
		arr[s[i]-'a'] = i
	}
	for i := 0 ; i < len(s);i++ {
		if arr[s[i]-'a'] == i {
			return i
		}
		arr[s[i]-'a'] = -1
	}
	return -1
}

File: OneQuestionPerDay/387/first_unique_character_in_a_string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        arr = [0] * 26

        for i, c in enumerate(s):
            arr[ord(c) - ord('a')] = i

        for i, c in enumerate(s):
            if arr[ord(c) - ord('a')] == i:
                return i
            arr[ord(c) - ord('a')] = -1
        return -1
